config/foreman-proxy-content.migrations/210407174237-add-puppet-reserved-code-cache.rb

```diff
@@ -0,0 +1,19 @@
+if answers['puppet'].is_a?(Hash)
+  reserved_code_cache_option = '-XX:ReservedCodeCacheSize'
+  reserved_code_cache_arg = "#{reserved_code_cache_option}=512m"
+  # Optional[Variant[String,Array[String]]]
```

Comment:
What does this comment mean in context here?

Reply:
The data type of the variable. It should tell you enough to understand why there are 3 cases.

Reply:
Ahh, consider updating the comment to mention that cause I did not piece those two together :)

Reply:
Since I've been [thinking about `Optional` a lot recently](https://github.com/alexjfisher/puppet-lint-optional_default-check), I'm wondering what I think of `Optional[Variant[..` .

Give that `Optional[Foo]` is nothing more than a shorthand for `Variant[Undef, Foo]`...

Instead of `Optional[Variant[String,Array[String]]]`, is `Variant[Undef,String,Array[String]]` better style??

Reply:
I have followed your interest and had the same thought. However, I now just copied it from the definition: https://github.com/theforeman/puppet-puppet/blob/c64b89694fb2a0815d1ade68d0cf98aa36524745/manifests/init.pp#L698

Reply:
Yep, just something that caught my eye.  No need to change anything.